radix-engine-interface/src/blueprints/resource/non_fungible/non_fungible_resource_manager.rs

```diff
@@ -304,30 +290,61 @@ pub struct NonFungibleResourceManagerMintSingleRuidTypedInput<T> {
 pub type NonFungibleResourceManagerMintSingleRuidOutput = (Bucket, NonFungibleLocalId);
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor, ManifestSbor)]
-pub struct NonFungibleDataSchema {
-    pub schema: VersionedScryptoSchema,
-    pub non_fungible: LocalTypeId,
-    pub mutable_fields: IndexSet<String>,
+pub enum NonFungibleDataSchema {
+    // TODO: ignore this variant in Scrypto for smaller code size
+    Local {
+        schema: VersionedScryptoSchema,
+        type_id: LocalTypeId,
+        mutable_fields: IndexSet<String>,
+    },
+    Remote {
+        type_id: BlueprintTypeId,
+        mutable_fields: IndexSet<String>,
+    },
 }
 
 impl NonFungibleData for () {
     const MUTABLE_FIELDS: &'static [&'static str] = &[];
 }
 
 impl NonFungibleDataSchema {
-    pub fn new_schema<N: NonFungibleData>() -> Self {
+    /// Arguments:
+    /// * [`package_address`] - The package address to use for resolving types defined within this package
+    pub fn new_local<N: NonFungibleData>(package_address: PackageAddress) -> Self {
         let mut aggregator = TypeAggregator::<ScryptoCustomTypeKind>::new();
-        let non_fungible_type = aggregator.add_child_type_and_descendents::<N>();
+        let type_id = aggregator.add_child_type_and_descendents::<N>();
+        let mut schema = generate_full_schema(aggregator);
+        replace_self_package_address(&mut schema, package_address);
```

Comment:
Can you double-check that we validate that the replacement has been done when validating the schema?

Reply:
Hmm I've checked in `custom_schema.rs` - looks like we don't do this validation

Reply:
But looks like this is resolved automatically to the blueprint in `matches_with_origin` anyway

Reply:
Ignore all this thread, as per discussions